refactor user routes and controllers

Rahul-spec-tech opened the pull request:
### Context
Moved business logic to controller files and separated route definitions into the routes folder for improved code organization and maintainability.

### Changes Introduced
- Refactored user-related routes to utilize controller functions.
- Moved all user business logic into the UserController.
- Defined user routes in the routes folder.

This refactor enhances the separation of concerns and simplifies future maintenance.

Commit:
auth md

Commit:
update

Commit:
Update auth.md

Commit:
Issue

Finding error while working with User API

Commit:
Refactor: Move business logic to controllers and define routes in routes folder

- Moved all business logic from routes to controller files.
- Defined route handling in the routes folder for better separation of concerns.

NithinSK04 commented on models/controllers/Post_Controller:
```diff
@@ -0,0 +1,75 @@
+    try {
+      const post = new Post({ title, description, author });
+      await post.save();
+      res.status(201).send(post);
```
res.status(201).json(post)

Commit:
Api Creation

Created the api for user and post

Commit:
Implement User and Post API with CRUD operations

- Added User API with endpoints for creating, reading, updating, and deleting user data.
- Added Post API with endpoints for creating, reading, updating, and deleting post data.
- Ensured proper validation and error handling for both User and Post models.
- Included necessary middleware for authentication and authorization.
- Updated database schemas for User and Post to reflect current requirements.
- Refactored code for better readability and maintainability.

Commit:
Add user authentication

- Updated the User model to include a password field and hash the password before saving it.
- Added user registration and login functionality in the user controller.
- Implemented bcrypt for password hashing and jsonwebtoken for generating tokens.
- Added a login route to the user routes.
- Created a middleware to verify JWT tokens.
- Protected post routes using the new authentication middleware.

Commit:
Update .gitignore

I was updated the .gitignore file where the node-modules are not uploaded while creating the pull request

Commit:
Add .gitignore to exclude node_modules and .env file

Added .gitignore file to exclude node_modules directory and .env file from version control.
The .env file contains sensitive information such as the MongoDB connection string and the application port, and should not be tracked by Git.

Commit:
Refactored post-controller to centralize JWT token handling and updated user-controller

Refactored post-controller to centralize JWT token handling, ensuring consistent user authorization. Updated user-controller to provide clearer error messages for duplicate user registration and other validation errors. Adjusted .env for better JWT secret and expiration time management.

Commit:
Updated JWT_SECRET_KEY

updated JWT_SECRET_KEY and the middleware of application

Commit:
Merge pull request #4 from Rahul-spec-tech/LoginPageAuthentication

Login page authentication
